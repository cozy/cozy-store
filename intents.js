!function(t){function e(e){for(var r,i,c=e[0],l=e[1],u=e[2],p=0,f=[];p<c.length;p++)i=c[p],o[i]&&f.push(o[i][0]),o[i]=0;for(r in l)Object.prototype.hasOwnProperty.call(l,r)&&(t[r]=l[r]);for(s&&s(e);f.length;)f.shift()();return a.push.apply(a,u||[]),n()}function n(){for(var t,e=0;e<a.length;e++){for(var n=a[e],r=!0,c=1;c<n.length;c++){var l=n[c];0!==o[l]&&(r=!1)}r&&(a.splice(e--,1),t=i(i.s=n[0]))}return t}var r={},o={3:0},a=[];function i(e){if(r[e])return r[e].exports;var n=r[e]={i:e,l:!1,exports:{}};return t[e].call(n.exports,n,n.exports,i),n.l=!0,n.exports}i.m=t,i.c=r,i.d=function(t,e,n){i.o(t,e)||Object.defineProperty(t,e,{enumerable:!0,get:n})},i.r=function(t){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(t,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(t,"__esModule",{value:!0})},i.t=function(t,e){if(1&e&&(t=i(t)),8&e)return t;if(4&e&&"object"==typeof t&&t&&t.__esModule)return t;var n=Object.create(null);if(i.r(n),Object.defineProperty(n,"default",{enumerable:!0,value:t}),2&e&&"string"!=typeof t)for(var r in t)i.d(n,r,function(e){return t[e]}.bind(null,r));return n},i.n=function(t){var e=t&&t.__esModule?function(){return t.default}:function(){return t};return i.d(e,"a",e),e},i.o=function(t,e){return Object.prototype.hasOwnProperty.call(t,e)},i.p="";var c=window.webpackJsonp=window.webpackJsonp||[],l=c.push.bind(c);c.push=e,c=c.slice();for(var u=0;u<c.length;u++)e(c[u]);var s=l;a.push([0,1,0]),n()}({0:function(t,e,n){n("201c"),t.exports=n("d/w2")},"d/w2":function(t,e,n){"use strict";n.r(e);n("VxdY");var r=n("/MKj"),o=n("q1tI"),a=n.n(o),i=n("i8i4"),c=n("MBKY"),l=n("XOpu"),u=n("zGKU");function s(t){return(s="function"==typeof Symbol&&"symbol"==typeof Symbol.iterator?function(t){return typeof t}:function(t){return t&&"function"==typeof Symbol&&t.constructor===Symbol&&t!==Symbol.prototype?"symbol":typeof t})(t)}function p(t,e){for(var n=0;n<e.length;n++){var r=e[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(t,r.key,r)}}function f(t,e){return!e||"object"!==s(e)&&"function"!=typeof e?function(t){if(void 0===t)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return t}(t):e}function y(t){return(y=Object.setPrototypeOf?Object.getPrototypeOf:function(t){return t.__proto__||Object.getPrototypeOf(t)})(t)}function b(t,e){return(b=Object.setPrototypeOf||function(t,e){return t.__proto__=e,t})(t,e)}var d=function(t){function e(t){var n;return function(t,e){if(!(t instanceof e))throw new TypeError("Cannot call a class as a function")}(this,e),(n=f(this,y(e).call(this,t))).state={error:null,service:null,status:"creating"},t.intents.createService().then(function(t){return n.setState({status:"created",service:t})}).catch(function(t){return n.setState({error:t,status:"errored"})}),n}return function(t,e){if("function"!=typeof e&&null!==e)throw new TypeError("Super expression must either be null or a function");t.prototype=Object.create(e&&e.prototype,{constructor:{value:t,writable:!0,configurable:!0}}),e&&b(t,e)}(e,o["Component"]),function(t,e,n){e&&p(t.prototype,e),n&&p(t,n)}(e,[{key:"render",value:function(){var t=this.props,e=t.appData,n=t.children,r=t.t,i=this.state,c=i.error,l=i.service,s=i.status,p=l&&l.getIntent(),f=p&&o.Children.toArray(n).find(function(t){var e=t.props,n=e.action,r=e.type;return n===p.attributes.action&&r===p.attributes.type});return a.a.createElement("div",{className:"coz-intent"},"creating"===s&&a.a.createElement(u.a,{size:"xxlarge"}),c&&a.a.createElement("div",{className:"coz-error coz-intent-error"},a.a.createElement("p",null,r("intent.service.creation.error.title")),a.a.createElement("p",null,c.message)),f&&a.a.cloneElement(f,{appData:e,data:l.getData(),intent:l.getIntent(),onCancel:function(){return l.cancel()},onError:function(t){return l.throw(t)},onTerminate:function(t){return l.terminate(t)}}))}}]),e}(),h=Object(l.e)()(d),m=n("PShk"),v=n("7Z11"),g=n("/n/j");function E(t){return(E="function"==typeof Symbol&&"symbol"==typeof Symbol.iterator?function(t){return typeof t}:function(t){return t&&"function"==typeof Symbol&&t.constructor===Symbol&&t!==Symbol.prototype?"symbol":typeof t})(t)}function O(t,e){for(var n=0;n<e.length;n++){var r=e[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(t,r.key,r)}}function j(t){return(j=Object.setPrototypeOf?Object.getPrototypeOf:function(t){return t.__proto__||Object.getPrototypeOf(t)})(t)}function w(t,e){return(w=Object.setPrototypeOf||function(t,e){return t.__proto__=e,t})(t,e)}function S(t){if(void 0===t)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return t}var _={alreadyInstalledError:"intent.install.error.installed",fetchError:"intent.install.error.fetch",installError:"intent.install.error.install",appError:"intent.install.error.unavailable"},P=function(t){function e(t){var n;return function(t,e){if(!(t instanceof e))throw new TypeError("Cannot call a class as a function")}(this,e),function(t,e,n){e in t?Object.defineProperty(t,e,{value:n,enumerable:!0,configurable:!0,writable:!0}):t[e]=n}(S(S(n=function(t,e){return!e||"object"!==E(e)&&"function"!=typeof e?S(t):e}(this,j(e).call(this,t)))),"state",{status:""}),t.initAppIntent(),n}return function(t,e){if("function"!=typeof e&&null!==e)throw new TypeError("Super expression must either be null or a function");t.prototype=Object.create(e&&e.prototype,{constructor:{value:t,writable:!0,configurable:!0}}),e&&w(t,e)}(e,o["Component"]),function(t,e,n){e&&O(t.prototype,e),n&&O(t,n)}(e,[{key:"installApp",value:function(){var t=this.props,e=t.app,n=t.data,r=t.isInstalling,o=t.isFetching,a=t.fetchError;if(!r&&!o&&!a)return this.props.installApp(n.slug,e.type)}},{key:"componentWillReceiveProps",value:function(t){this.props.isInstalling&&!t.isInstalling&&(this.props.onTerminate(t.app),this.setState({status:"installed"}))}},{key:"render",value:function(){var t=this,e=this.props,n=e.app,r=e.appData,o=e.fetchError,i=e.installError,c=e.isFetching,l=e.isAppFetching,s=e.isInstalling,p=e.onCancel,f=e.t,y=this.state.status,b=c||l,d=!o&&!b&&!l,h="installed"===y,g=d&&!n,E={alreadyInstalledError:!h&&n&&n.installed,fetchError:o,installError:i,appError:g},O=Object.keys(E).reduce(function(t,e){return E[e]?_[e]:t},null),j=!!O&&new Error(f(O));return a.a.createElement("div",{className:"coz-intent-wrapper"},a.a.createElement(v.a,{appEditor:r.cozyAppEditor,appName:r.cozyAppName,appIcon:"../".concat(r.cozyIconPath)}),a.a.createElement("div",{className:"coz-intent-content"},b&&a.a.createElement(u.a,{size:"xxlarge"}),j&&a.a.createElement("div",{className:"coz-error"},j.message),d&&!h&&!j&&a.a.createElement(m.a,{appSlug:n.slug,installApp:function(){return t.installApp()},isInstalling:s,onCancel:p})))}}]),e}(),z=Object(l.e)()(Object(r.b)(function(t,e){return{isFetching:t.apps.isFetching,isAppFetching:t.apps.isAppFetching,isInstalling:t.apps.isInstalling,installError:t.apps.actionError,fetchError:t.apps.fetchError,app:Object(g.f)(t,e.data.slug)}},function(t,e){return{initAppIntent:function(){t(Object(g.k)(e.lang,e.data.slug))},installApp:function(e,n,r){return t(Object(g.l)(e,n,r))}}})(P));document.addEventListener("DOMContentLoaded",function(){var t=document.querySelector("[role=application]").dataset;cozy.client.init({cozyURL:"//"+t.cozyDomain,token:t.cozyToken}),Object(i.render)(a.a.createElement(l.c,{lang:t.cozyLocale,dictRequire:function(t){return n("/KVF")("./".concat(t))}},a.a.createElement(r.a,{store:c.a},a.a.createElement(h,{appData:t,intents:cozy.client.intents},a.a.createElement(z,{action:"INSTALL",type:"io.cozy.apps"})))),document.querySelector("[role=application]"))})}});