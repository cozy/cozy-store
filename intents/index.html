<!DOCTYPE html><html lang="{{.Locale}}"><head><meta charset="utf-8"><title>Cozy Store intents</title><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" href="/favicon-32x32.png" sizes="32x32"><link rel="icon" type="image/png" href="/favicon-16x16.png" sizes="16x16"><link rel="manifest" href="/site.webmanifest"><link rel="stylesheet" type="text/css" href="//{{.Domain}}/assets/fonts/fonts.css"><meta name="theme-color" content="#ffffff"><meta name="viewport" content="width=device-width,initial-scale=1"><link rel="stylesheet" href="../intents.1a9c4c91daa33f36fa65.min.css"><script src="../intents.js" defer="defer"></script>{{.CozyClientJS}}</head><div role="application" data-cozy-token="{{.Token}}" data-cozy-domain="{{.Domain}}" data-cozy-locale="{{.Locale}}" data-cozy-app-editor="{{.AppEditor}}" data-cozy-app-name="{{.AppName}}" data-cozy-app-name-prefix="{{.AppNamePrefix}}" data-cozy-icon-path="{{.IconPath}}"></div></html>