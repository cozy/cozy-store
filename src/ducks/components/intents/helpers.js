export const isPermissionsPageToDisplay = data => {
  const { pageToDisplay, category, qualificationLabels, slugList } = data || {}
  return (
    !category &&
    !qualificationLabels &&
    !slugList &&
    (!pageToDisplay || pageToDisplay === 'permissions')
  )
}
