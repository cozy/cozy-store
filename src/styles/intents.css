div[role=application] {
  width: 100vw;
  height: 100vh;
}

.sto-intent {
    width: 100%;
    height: 100%;
}

.sto-intent-content {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    overflow-y: auto;
}

.sto-intent-content .sto-modal-content {
  margin-top: 3rem;
  max-height: calc(100% - 3rem);
}

.sto-intent-content--loading {
  display: flex;
  flex-direction: column;
  justify-content: center;
}
