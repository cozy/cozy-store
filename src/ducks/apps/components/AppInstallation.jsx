import React, { Component } from 'react'

import { connect } from 'react-redux'
import { PropTypes } from 'react-proptypes'
import flags from 'cozy-flags'

import ReactMarkdownWrapper from 'ducks/components/ReactMarkdownWrapper'
import { ModalDescription, ModalHeader, ModalFooter } from 'cozy-ui/react/Modal'
import Spinner from 'cozy-ui/react/Spinner'
import Button from 'cozy-ui/react/Button'

import PermissionsList from 'ducks/apps/components/PermissionsList'
import Partnership from 'ducks/apps/components/Partnership'
import { translate } from 'cozy-ui/react/I18n'
import Alerter from 'cozy-ui/react/Alerter'
import Checkbox from 'cozy-ui/react/Checkbox'

import { getTranslatedManifestProperty } from 'lib/helpers'
import { hasPendingUpdate } from 'ducks/apps/appStatus'
import storeConfig from 'config'

import { APP_TYPE, getAppBySlug, installAppFromRegistry } from 'ducks/apps'

const shouldSkipPermissions = app =>
  flags('skip-low-permissions') &&
  app.label <= storeConfig.default.authorizedLabelLimit

const hasInstallation = app => !app.installed || hasPendingUpdate(app)

export class AppInstallation extends Component {
  constructor(props) {
    super(props)
    const { app } = props
    this.state = { isTermsAccepted: false }
    this.installApp = this.installApp.bind(this)
    if (shouldSkipPermissions(app) && hasInstallation(app)) {
      this.installApp()
    }
  }

  installApp() {
    if (!this.isInstallReady()) return // Not ready to be installed
    const { app, channel, installApp, onInstallOrUpdate, t } = this.props
    const isUpdate = app.installed
    const isChannelSwitch = !!channel
    installApp(app, channel, isUpdate).catch(error => {
      const appName = getTranslatedManifestProperty(app, 'name', t)
      Alerter.error(
        t('app_modal.install.alert.install_error', {
          message: error.message,
          name: appName
        }),
        {
          duration: 12000,
          buttonText: t('app_modal.install.alert.dismiss'),
          buttonAction: dismiss => dismiss()
        }
      )
      console.error(error)
    })
    if (isUpdate || isChannelSwitch || app.type !== APP_TYPE.KONNECTOR) {
      if (typeof onInstallOrUpdate === 'function') onInstallOrUpdate()
    }
  }

  componentDidUpdate = prevProps => {
    const { app, onKonnectorInstall } = this.props
    const justInstalled =
      !!prevProps.app && !prevProps.app.installed && app.installed
    if (justInstalled && app.type === APP_TYPE.KONNECTOR) {
      if (typeof onKonnectorInstall === 'function') onKonnectorInstall()
    }
  }

  acceptTerms = () => {
    this.setState(state => ({
      isTermsAccepted: !state.isTermsAccepted
    }))
  }

  isInstallReady = () => {
    const { app, isInstalling } = this.props
    const isCurrentAppInstalling = isInstalling === app.slug
    const isTermsReady =
      shouldSkipPermissions(app) || (!app.terms || this.state.isTermsAccepted)
    return !isCurrentAppInstalling && isTermsReady
  }

  render() {
    const {
      app,
      fetchError,
      installError,
      isFetching,
      isAppFetching,
      isInstalling,
      onCancel,
      t
    } = this.props
    const { isTermsAccepted } = this.state
    const isFetchingSomething = isFetching || isAppFetching
    const isCurrentAppInstalling = isInstalling === app.slug

    if (shouldSkipPermissions(app) && hasInstallation(app)) {
      return (
        <React.Fragment>
          <ModalDescription className="sto-install-loading-wrapper">
            <div className="sto-install-loading">
              <Spinner size="xxlarge" />
            </div>
          </ModalDescription>
        </React.Fragment>
      )
    }

    return (
      <React.Fragment>
        <ModalHeader title={t('app_modal.install.title')} />
        {isFetchingSomething ? (
          <ModalDescription className="sto-install-loading-wrapper">
            <div className="sto-install-loading">
              <Spinner size="xlarge" />
            </div>
          </ModalDescription>
        ) : (
          <ModalDescription>
            {app.partnership && <Partnership app={app} />}
            {app.permissions && <PermissionsList app={app} />}
            {fetchError && (
              <p className="u-error">
                {t('app_modal.install.message.version_error', {
                  message: fetchError.message
                })}
              </p>
            )}
          </ModalDescription>
        )}
        {!isFetchingSomething && !fetchError && (
          <ModalFooter className="sto-install-footer">
            {installError && (
              <p className="u-error">
                {t('app_modal.install.message.install_error', {
                  message: installError.message
                })}
              </p>
            )}
            {app.terms && (
              <div className="sto-install-terms">
                <Checkbox
                  className="sto-install-terms-checkbox"
                  onChange={this.acceptTerms}
                  checked={isTermsAccepted}
                  disabled={isInstalling}
                >
                  <ReactMarkdownWrapper
                    source={t('app_modal.install.terms', {
                      url: app.terms.url
                    })}
                  />
                </Checkbox>
              </div>
            )}
            <div className="sto-install-controls">
              <Button
                theme="secondary"
                onClick={onCancel}
                disabled={isCurrentAppInstalling}
                label={t('app_modal.install.cancel')}
                extension="full"
                className="u-mh-half"
              />
              <Button
                theme="primary"
                disabled={!this.isInstallReady() || isInstalling}
                busy={isCurrentAppInstalling}
                extension="full"
                onClick={this.installApp}
                label={
                  hasPendingUpdate(app)
                    ? t('app_modal.install.update')
                    : t('app_modal.install.install')
                }
                className="u-mh-half"
              />
            </div>
          </ModalFooter>
        )}
      </React.Fragment>
    )
  }
}

AppInstallation.propTypes = {
  appSlug: PropTypes.string.isRequired
}

const mapStateToProps = (state, ownProps) => ({
  app: getAppBySlug(state, ownProps.appSlug),
  isFetching: state.apps.isFetching,
  isAppFetching: state.apps.isAppFetching,
  isInstalling: state.apps.isInstalling,
  fetchError: state.apps.fetchError,
  installError: state.apps.actionError
})

const mapDispatchToProps = dispatch => ({
  installApp: (app, channel, isUpdate) =>
    dispatch(installAppFromRegistry(app, channel, isUpdate))
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(translate()(AppInstallation))
