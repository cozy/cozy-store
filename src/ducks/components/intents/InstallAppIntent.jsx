import React, { Component } from 'react'
import { connect } from 'react-redux'
import { translate } from 'cozy-ui/react/I18n'

import AppInstallation from '../../apps/components/AppInstallation'
import IntentHeader from 'cozy-ui/react/IntentHeader'
import Spinner from 'cozy-ui/react/Spinner'

import { getAppBySlug, installAppFromRegistry, initAppIntent } from '../../apps'

const errorKeys = {
  alreadyInstalledError: 'intent.install.error.installed',
  fetchError: 'intent.install.error.fetch',
  installError: 'intent.install.error.install',
  appError: 'intent.install.error.unavailable'
}

export class InstallAppIntent extends Component {
  state = {
    status: ''
  }

  constructor(props) {
    super(props)
    props.initAppIntent()
  }

  installApp() {
    const { app, data, isInstalling, isFetching, fetchError } = this.props
    const isReady = !isInstalling && !isFetching && !fetchError

    if (isReady) {
      return this.props.installApp(data.slug, app.type)
    }
  }

  componentWillReceiveProps(nextProps) {
    // on install success
    if (this.props.isInstalling && !nextProps.isInstalling) {
      this.props.onTerminate(nextProps.app)
      this.setState({
        status: 'installed'
      })
    }
  }

  render() {
    const {
      app,
      appData,
      fetchError,
      installError,
      isFetching,
      isAppFetching,
      isInstalling,
      onCancel,
      t
    } = this.props

    const { status } = this.state

    const fetching = isFetching || isAppFetching
    const isReady = !fetchError && !fetching && !isAppFetching
    const isInstalled = status === 'installed'

    const appError = isReady && !app
    const alreadyInstalledError = !isInstalled && app && app.installed
    const errors = { alreadyInstalledError, fetchError, installError, appError }
    const errorKey = Object.keys(errors).reduce(
      (final, key) => (errors[key] ? errorKeys[key] : final),
      null
    )
    const error = !!errorKey && new Error(t(errorKey))

    return (
      <div className="coz-intent-wrapper">
        <IntentHeader
          appEditor={appData.cozyAppEditor}
          appName={appData.cozyAppName}
          appIcon={`../${appData.cozyIconPath}`}
        />
        <div className="coz-intent-content">
          {fetching && <Spinner size="xxlarge" />}
          {error && <div className="coz-error">{error.message}</div>}
          {isReady &&
            !isInstalled &&
            !error && (
              <AppInstallation
                appSlug={app.slug}
                installApp={() => this.installApp()}
                isInstalling={isInstalling}
                onCancel={onCancel}
              />
            )}
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => ({
  isFetching: state.apps.isFetching,
  isAppFetching: state.apps.isAppFetching,
  isInstalling: state.apps.isInstalling,
  installError: state.apps.actionError,
  fetchError: state.apps.fetchError,
  app: getAppBySlug(state, ownProps.data.slug)
})

const mapDispatchToProps = (dispatch, ownProps) => ({
  initAppIntent: () => {
    dispatch(initAppIntent(ownProps.lang, ownProps.data.slug))
  },
  installApp: (appSlug, appType, channel) =>
    dispatch(installAppFromRegistry(appSlug, appType, channel))
})

// translate last to pass the lang property to fetchApps()
export default translate()(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(InstallAppIntent)
)
