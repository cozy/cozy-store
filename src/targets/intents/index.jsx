import 'styles'

import { Provider } from 'react-redux'
import React from 'react'
import { render } from 'react-dom'
import store from 'lib/store'

import CozyClient, { CozyProvider } from 'cozy-client'
import I18n from 'cozy-ui/transpiled/react/I18n'
import flag from 'cozy-flags'

import IntentHandler from '../../ducks/components/intents/IntentHandler'
import InstallAppIntent from '../../ducks/components/intents/InstallAppIntent'
import schema from 'lib/schema'

document.addEventListener('DOMContentLoaded', () => {
  const root = document.querySelector('[role=application]')
  const appData = root.dataset

  const protocol = window.location.protocol
  const client = new CozyClient({
    uri: `${protocol}//${appData.cozyDomain}`,
    schema,
    token: appData.cozyToken,
    store: false
  })
  client.registerPlugin(flag.plugin)

  render(
    <I18n
      lang={appData.cozyLocale}
      dictRequire={lang => require(`../../locales/${lang}`)}
    >
      <CozyProvider store={store} client={client}>
        <Provider store={store}>
          <IntentHandler appData={appData}>
            <InstallAppIntent action="INSTALL" type="io.cozy.apps" />
          </IntentHandler>
        </Provider>
      </CozyProvider>
    </I18n>,
    document.querySelector('[role=application]')
  )
})
