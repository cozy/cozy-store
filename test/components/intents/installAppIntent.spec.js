/* eslint-env jest */

import React from 'react'
import { mount, shallow } from 'enzyme'

import { InstallAppIntent } from 'ducks/components/intents/InstallAppIntent'
import AppInstallation from 'ducks/apps/components/deprecated/AppInstallation'
import CozyClient, { CozyProvider } from 'cozy-client'

jest.mock('ducks/apps/components/deprecated/AppInstallation')

AppInstallation.mockImplementation(() => null)

describe('InstallAppIntent component', () => {
  const props = {
    app: {
      slug: 'drive'
    },
    appData: {
      app: {
        editor: 'Cozy',
        name: 'Drive',
        icon: '/path/to/icon'
      }
    },
    initAppIntent: jest.fn()
  }

  afterEach(() => {
    jest.clearAllMocks()
  })

  const client = new CozyClient({})

  it('should be rendered correctly', () => {
    const componentWrapper = shallow(
      <CozyProvider client={client}>
        <InstallAppIntent {...props} />
      </CozyProvider>
    )
      .find(InstallAppIntent)
      .dive()
    const component = componentWrapper.getElement()

    expect(component).toMatchSnapshot()
  })

  it('should render AppInstallation correctly', () => {
    const root = mount(
      <CozyProvider client={client}>
        <InstallAppIntent {...props} />
      </CozyProvider>
    )

    expect(root.find(AppInstallation).length).toBe(1)
  })
})
