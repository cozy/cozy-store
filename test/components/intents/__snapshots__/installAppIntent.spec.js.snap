// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallAppIntent component should be rendered correctly 1`] = `
<div
  className="coz-intent-wrapper"
>
  <IntentHeader
    appEditor="Cozy"
    appIcon="..//path/to/icon"
    appName="Drive"
  />
  <div
    className="coz-intent-content"
  >
    <withClient(Connect(Wrapper))
      appSlug="drive"
      installApp={[Function]}
    />
  </div>
</div>
`;
